searx/engines/invidious.py

```diff
@@ -19,8 +20,8 @@
 time_range_support = True
 
 # search-url
-base_url = "https://invidio.us/"
-
+base_url = ["invidious.tube", "invidious.snopyta.org", "invidiou.site"]
+base_url_rand = random.choice(base_url)
```

Comment:
As it is written, an instance is picked randomly when searx starts, the other instances are ignored as long as searx is not restarted.

If `base_url` is a string, `base_url_rand` will contains `h` (the first character of `base_url`).

Reply:
> As it is written, an instance is picked randomly when searx starts, the other instances are ignored as long as searx is not restarted.

Yeah, i didn't realize this, i though it will pick a different instance for each request...

> If `base_url` is a string, `base_url_rand` will contains `h` (the first character of `base_url` which most probably start with `https://` ).

What do you mean by `h` ?

Reply:
> What do you mean by h ?

Hum, actually, not `h` for sure.

But try this:
```python
base_url = "invidious.tube"
base_url_rand = random.choice(base_url)
```

You can see that `base_url_rand` contains only one character, because `random.choice(base_url)` sees `base_url` as an iterator of character : 

```python
for i in  ["invidious.tube", "invidious.snopyta.org", "invidiou.site"]:
  print(i)
for i in "invidious.tube":
  print(i)
```

#### Why we can `base_url = "invidious.tube"` ?

Because we can have this settings.yml: 
```yaml
  - name : invidious
    engine : invidious
    base_url : 'https://invidious.tube/'
    shortcut: iv
    timeout : 5.0
    disabled : True
```

#### The solution I see:
* remove `base_url = ["invidious.tube", "invidious.snopyta.org", "invidiou.site"]`
* remove line 24 `base_url_rand = random.choice(base_url)`.
* before line 34 ( `    search_url = "https://" + base_url_rand + "/api/v1/search?q={query}"` ):
```
if isinstance(base_url, list):
  base_url_rand = random.choice(base_url)
else:
  base_url_rand = base_url
```
* change settings.yml this way:
```yaml
  - name : invidious
    engine : invidious
    base_url : 
        - https://invidious.tube
        - https://invidious.snopyta.org
        - https://invidiou.site
    shortcut: iv
    timeout : 5.0
    disabled : True
```

searx uses a different instance on each request. The invidious instances are listed in settings.yml instead of the engine code.

So:
* the admin can send request to .onion ( `base_url: http://wgegijigj.onion` )
*  the admin can optionally do this:
```yaml
  - name : invidious
    engine : invidious
    base_url : https://invidious.tube
    shortcut: iv
    timeout : 5.0
    disabled : True
```

Reply:
Great ! Thx @dalf i'll take a look soon and follow your advice.
Many thanks, again !

Reply:
I’ve tried what you propose @dalf, but it doesn't work. 
There is no error in the result page neither in terminal. I just have no results whatever I’m looking for, with the "Sorry, there is no result..." message.

I tried to remove "https://" from `search_url = "https://" + base_url_rand + "/api/v1/search?q={query}"` because the list of url in settings.yml already contains https://, for the same result.

How can i see what search url is used ?

By the way, there is another problem with Invidious in general: even if an instance seems to be alive, the search function on that instance does not return any results...